docs/end_to_end_tutorials/dev_practices/e2e_evaluation.md

```diff
@@ -0,0 +1,45 @@
+# End-to-End Evaluation
+## Setting up an Evaluation Set
+
+To help with this, we've created some tools that automatically generate a dataset for you given a set of documents to query. (See examples below).
+
+In the future, we will also be able to create datasets automatically against tools.
+
+## Qualitative v.s. Quantitative Eval
+
+Quantitative eval is more useful when evaluating applications where there is a correct answer - for instance, validating that the choice of tools and their inputs are correct given the plan, or retrieving specific pieces of information, or attempting to produce intermediate output of a certain schema (e.g. JSON fields).
+
+Qualitative eval is more useful when generating long-form responses that are meant to be *helpful* but not necessarily completely accurate.
+
+
+## Discovery - Sensitivity Testing
+
+With a complex pipeline, it may be unclear which parts of the pipeline are affecting your results.
+
+Sensitivity testing can be a good inroad into choosing which components to individually test or tweak more thoroughly.
+
+More details on how to discover issues automatically with methods such as sensitivity testing will come soon.
+
+## Metrics Ensembling
```

Comment:
we already have a bunch of core eval modules, can we make sure they're highlighted as sections in this page somehow? https://gpt-index.readthedocs.io/en/latest/core_modules/supporting_modules/evaluation/root.html